framework/src/functions/MooseParsedFunctionBase.C

```diff
@@ -36,11 +37,19 @@ MooseParsedFunctionBase::MooseParsedFunctionBase(const InputParameters & paramet
   if (_vars.size() != _vals.size())
     mooseError("Number of vars must match the number of vals for a MooseParsedFunction!");
 
-  // Loop through the variables assigned by the user and give an error if x,y,z,t are used
   for (const auto & var : _vars)
+  {
+    // Check for non-linear variables.
+    if (_pfb_feproblem.getNonlinearSystemBase().hasVariable(var))
```

Comment:
need to check the auxiliary system too

Reply:
Got it, if both need to be banned then the standard "hasVariable" already checks both systems so I will just do it that way